pkg/controller/vitesskeyspace/vitesskeyspace_controller.go

```diff
@@ -198,3 +220,18 @@ func (r *ReconcileVitessKeyspace) Reconcile(request reconcile.Request) (reconcil
 	reconcileCount.WithLabelValues(vtk.Labels[planetscalev2.ClusterLabel], vtk.Spec.Name, metrics.Result(err)).Inc()
 	return result, err
 }
+
+// newWrangler initializes a new Vitess Wrangler that gives us access to information
+// about resharding workflows.
+func newWrangler(ctx context.Context, vtk planetscalev2.VitessKeyspace) (*wrangler.Wrangler, error) {
```

Comment:
If we don't pass a pointer for the vtk, it will copy it, and it's a large object: 

```suggestion
func newWrangler(ctx context.Context, vtk *planetscalev2.VitessKeyspace) (*wrangler.Wrangler, error) {
```

Reply:
Also, I think it would be better to scope the inputs to exactly what we need rather than passing the entire vtk, so scratch my last comment. 
```suggestion
func newWrangler(ctx context.Context, lockserverSpec planetscalev2.LockserverSpec) (*wrangler.Wrangler, error) {
```

Reply:
I think you mean `planetscalev2.VitessLockserverParams` right? But that makes sense - done.

Reply:
Yep